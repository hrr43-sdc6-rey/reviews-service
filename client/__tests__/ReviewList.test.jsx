import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import ReviewList from '../components/ReviewList.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('ReviewList', () => {
  const reviewsArr = [
    {
      id: 77,
      experience_id: 5,
      username: 'Bessie',
      avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/anasnakawa/128.jpg',
      date: '2020-02-01T08:00:00.000Z',
      stars: 2,
      review: 'Commodi atque sit eos sit voluptas sint soluta numquam harum. Sed occaecati perferendis accusantium cum aut placeat ex dolor. Molestiae dolores eum. Omnis impedit illum tempore nihil consequatur rerum aut maiores. Eligendi fuga ut nobis odit porro sit dignissimos laborum. Natus cupiditate et.',
    },
    {
      id: 214,
      experience_id: 5,
      username: 'Stanley',
      avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/alagoon/128.jpg',
      date: '2020-01-01T08:00:00.000Z',
      stars: 4,
      review: 'Dolorum vero possimus hic sint ut. Ut et perspiciatis voluptate exercitationem tempora laboriosam totam suscipit et. Quia magni saepe eos.',
    },
  ];
  const wrapper = shallow(<ReviewList reviews={reviewsArr} />);

  it('ReviewList should render', () => {
    expect(wrapper).toBeDefined();
  });

  it('ReviewList snapshot should match shallow render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
