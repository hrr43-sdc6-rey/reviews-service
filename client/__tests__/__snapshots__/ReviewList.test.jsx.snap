// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList ReviewList snapshot should match shallow render 1`] = `
<div>
  <div
    className="reviewListItems"
  >
    <ReviewListItem
      idx={0}
      key="77"
      month="February"
      review={
        Object {
          "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/anasnakawa/128.jpg",
          "date": "2020-02-01T08:00:00.000Z",
          "experience_id": 5,
          "id": 77,
          "review": "Commodi atque sit eos sit voluptas sint soluta numquam harum. Sed occaecati perferendis accusantium cum aut placeat ex dolor. Molestiae dolores eum. Omnis impedit illum tempore nihil consequatur rerum aut maiores. Eligendi fuga ut nobis odit porro sit dignissimos laborum. Natus cupiditate et.",
          "stars": 2,
          "username": "Bessie",
        }
      }
      stars={2}
      year="2020"
    />
    <ReviewListItem
      idx={1}
      key="214"
      month="January"
      review={
        Object {
          "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/alagoon/128.jpg",
          "date": "2020-01-01T08:00:00.000Z",
          "experience_id": 5,
          "id": 214,
          "review": "Dolorum vero possimus hic sint ut. Ut et perspiciatis voluptate exercitationem tempora laboriosam totam suscipit et. Quia magni saepe eos.",
          "stars": 4,
          "username": "Stanley",
        }
      }
      stars={4}
      year="2020"
    />
  </div>
</div>
`;
